### PR TITLE
Added User default for the temperature unit C or F

### PR DESCRIPTION
**Pull request by clayton256**
After about 15 years, I want restart learning some Mac programming. I thought I would add this feature. Do you find it worthy to add? Is there a better way to do this? Can you make a recommendations? Thank you.

### CODE REVIEW

**Comment by clayton256** [comment]
I got too many files in the PR. Let me try again in the morning.

**Comment by uthiel** [comment]
not sure why you closed this PR, but here are my thoughts on it:

- instead of letting the user manually chose the temperature unit, store it in the user prefs and do some manual calculations, you should use the [NSMeasurement API](https://developer.apple.com/documentation/foundation/nsmeasurementformatter/1642057-stringfrommeasurement?language=objc) which does all of this in one call. Formatters are quite expensive to instantiate, so one would preferably retain it in an ivar.
- do all of this in one place, instead of scattered over multiple classes.

Let me know if you want to submit another PR or have me add it to the repo.

**Comment by clayton256** [comment]
I closed the PR because I accidentally included unnecessary files in it and couldn't figure out how to remove them. It was my first time trying to make git commits from Xcode vs the command line.
Thank you very much for your suggestion of the NSMesurement API. Of course Apple would provide this code!!
Does this look better? If so I'll submit a new PR:
```
- (NSString *)formattedTempForTemp:(double)temp {
    NSMeasurementFormatter *formatter = [NSMeasurementFormatter new];
    formatter.locale = NSLocale.autoupdatingCurrentLocale;
    formatter.unitStyle = NSFormattingUnitStyleMedium;
    return [formatter stringFromMeasurement:[[NSMeasurement alloc] initWithDoubleValue:temp unit:[NSUnitTemperature celsius]]];
}
```

**Comment by uthiel** [comment]
You're instantiating `NSFormatter` on every call. As I said before, `NSFormatter` is an expensive class to instantiate. You should therefore retain it in an ivar.
Looking forward to your PR.
